### ISSUE TITLE
core_aten_decompositions table have non-functional or CIA entries. 

### ISSUE THREAD

**Issue by tugsbayasgalan** [pytorch/pytorch#135759]
### 🐛 Describe the bug

core_aten_decompositions table is meant to be used in export context where we should assume every key is a functional op as the table would only be applied after functionalization is run. As a result, any non-functional entries should result in no-op. So we should remove them and implement tighter checks around core_aten_decompositions().  

### Versions

main

**Comment by zou3519** [comment]
The alternative is that we run decompositions above functionalization. This should be simple to implement: we can invoke the decomposition table in FunctionalTensor (instead of at ProxyMode).

**Comment by tugsbayasgalan** [comment]
> The alternative is that we allow for this behavior (the presence of in-place operations in decompositions) and run decompositions above functionalization. This should be simple to implement: we can invoke the decomposition table in FunctionalTensor (instead of at ProxyMode).

Yeah that is one option if we want to ever return customized non-functional inference IR? Or even if we want functional IR, your option gives more power to user to customize the IR now that it gives control to users on non-functional ops.

**Comment by bdhirsh** [comment]
that is true (in fact, it would fix some existing problems, see https://github.com/pytorch/pytorch/issues/129418).

 I am a bit curious if this will materially change any of the graphs that we see in compile today (given that inductor has a bunch of mutable ops in its decomp table that are no-ops today). But it feels pretty reasonable to do this (we can also just kick off a perf run on a PR that does it).
